sdk/resource/detect/auto.go

```diff
@@ -0,0 +1,44 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package detect
+
+import (
+	"context"
+
+	"go.uber.org/multierr"
```

Comment:
I'd rather not include `"go.uber.org/multierr"` in the project. The MIT license can be troublesome for dependencies and it seems like the functionality we are using it for is something we can handle natively.

Reply:
I see. The reason that we use this is because we want to combine errors. I find an alternative package https://github.com/pkg/errors#adding-context-to-an-error. So do we use the new package or change the speculation of the function to return the first error we met?

Reply:
I usually solve this class of problem using Go primative. Something like:

```go
var errs []string
for _, i := range something {
	if _, err := i.f(); err != nil {
		errs = append(errs, err.Error())
	}
	...
}
err := fmt.Errorf("iterating something:\n\t%s", strings.Join(errs, "\n\t"))
...
}
```

If more is needed from the errors make `errs` a `[]error` and process them as errors.

Reply:
For future reference: I was mistaking MPL-2.0 with MIT here which is the licensing for the alternate form of this package (github.com/hashicorp/go-multierror).